packages/next/src/client/components/react-dev-overlay/server/middleware.ts

```diff
@@ -241,9 +284,30 @@ export async function getSource(
   for (const compilation of getCompilations()) {
     // TODO: `ignoreList`
     const sourceMap = await getSourceMapFromCompilation(moduleId, compilation)
+    const ignoreList = []
```

Comment:
as a follow-up: Since we control the sourcemap, we can convert it to a decoded sourcemap immediately. That means we no longer have `ignoreList` or `sourcesContent` arrays but a list of "decoded sources". Each decoded sources holds properties for whether it's ignored and what its source content is. That way you don't have to do another O(N^2) lookup on the frontend to check whether a sources entry is ignore listed.

So from `{ sources: string[], sourcesContent: string[], ignoreList: number[] }` we go to `{ sources: Array<{ url: string, ignored: boolean, content: string | null }>}`

Reply:
I'll add a TODO for that. I also changed the loop a bit currently it's O(n) since I'm not using indexOf to get the index

Reply:
>  raw source map to the consumer later,

Why can't we keep the original source map payload and a decoded source map and pass it where appropriate. Then we can later check if we really need the raw source map. And "need" as in "we need this due to requirements" not due to the internal factoring of the code.

Reply:
`content` is large that's why I removed if we don't have to duplicate, can you explain why we can drop the raw source map possibly. I also kept the original source, just don't see why we need to have two copies of content yet. Unless we're clear where we gonna use them later